diff --git a/lisp/cus-start.el b/lisp/cus-start.el
index d9ad0a5..9e52d0f 100644
--- a/lisp/cus-start.el
+++ b/lisp/cus-start.el
@@ -446,6 +446,7 @@ minibuffer-prompt-properties--setter
 	     (ns-use-native-fullscreen ns boolean "24.4")
              (ns-use-fullscreen-animation ns boolean "25.1")
              (ns-use-srgb-colorspace ns boolean "24.4")
+             (ns-use-title-bar ns boolean "26.1")
 	     ;; process.c
 	     (delete-exited-processes processes-basics boolean)
 	     ;; syntax.c
diff --git a/src/nsterm.m b/src/nsterm.m
index 1b44a73..d013101 100644
--- a/src/nsterm.m
+++ b/src/nsterm.m
@@ -6775,11 +6775,16 @@ - (BOOL)isOpaque
   maximizing_resize = NO;
 #endif

-  win = [[EmacsWindow alloc]
+ #if MAC_OS_X_VERSION_MAX_ALLOWED < 101200
+ #define NSWindowStyleMaskBorderless NSBorderlessWindowMask
+ #endif
+  win = [[EmacsFSWindow alloc]
             initWithContentRect: r
                       styleMask: (NSResizableWindowMask |
 #if MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_7
-                                  NSTitledWindowMask |
+                                  (ns_use_title_bar ?
+                                   NSTitledWindowMask :
+                                   NSWindowStyleMaskBorderless) |
 #endif
                                   NSMiniaturizableWindowMask |
                                   NSClosableWindowMask)
@@ -6812,6 +6814,7 @@ - (BOOL)isOpaque
   [win setTitle: name];

   /* toolbar support */
+  if ( ns_use_title_bar ) {
   toolbar = [[EmacsToolbar alloc] initForView: self withIdentifier:
                          [NSString stringWithFormat: @"Emacs Frame %d",
                                    ns_window_num]];
@@ -6833,6 +6836,7 @@ This avoids an extra clear and redraw (flicker) at frame creation.  */
   }
 #endif
   FRAME_TOOLBAR_HEIGHT (f) = 0;
+  }

   tem = f->icon_name;
   if (!NILP (tem))
@@ -8759,6 +8763,17 @@ Nil means use fullscreen the old (< 10.7) way.  The old way works better with
 This variable is ignored on OSX < 10.7 and GNUstep.  */);
   ns_use_srgb_colorspace = YES;

+  DEFVAR_BOOL ("ns-use-title-bar", ns_use_title_bar,
+     doc: /*Non-nil to include the title bar on frames.
+Nil to omit the title bar on OSX >= 10.7. This variable is
+ignored on OSX < 10.7. Note that the first frame per
+session (unless that frame is an emacsclient) will ignore (!) the
+user-set value of this variable, and instead use the default
+value of this variable (nil--do not include a title bar). This is
+almost certainly due to a bug in the patch adding this variable.
+Default is nil.*/);
+  ns_use_title_bar = NO;
+
   /* TODO: move to common code */
   DEFVAR_LISP ("x-toolkit-scroll-bars", Vx_toolkit_scroll_bars,
 	       doc: /* Which toolkit scroll bars Emacs uses, if any.
